### PR TITLE
add Filter component #1847

### PR DESCRIPTION
**Pull request by pingan1927**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1918)

### CODE REVIEW

**Comment by pingan1927** [comment]
@silentcloud @paranoidjk 有空就CR下

File: components/filter/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,183 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (warmhug) [inline components/filter/__tests__/__snapshots__/demo.test.js.snap]: pr 里不必更新 snapshot ，review 多余、也容易产生冲突

**Comment by warmhug** [comment]
在 Flex 组件里 做一个这样的 demo ，不要做成单独组件。

**Comment by pingan1927** [comment]
> 在 Flex 组件里 做一个这样的 demo ，不要做成单独组件。

做成组件的意义在于降低重复开发的繁琐，如果写成Flex的一个demo，那实际上跟没有一样，类似的需求做过很多，每次都需要现找icon，而且都不一样，也存在一些同学随便就搞一个三角的字符的情况，完全没有规范性，不断地被要求完善成符合视觉规范。

**Comment by warmhug** [comment]
这个类似 支付宝6位密码框、业务属性太强，适合抽象到业务层面，而不是放到通用 UI 库里。
只提供 demo 确实传达效果不好，想想其他有什么更好办法？

**Comment by pingan1927** [comment]
这个还真没什么好办法，各业务线自己实现吧只有

**Pull request closed without merging**